ansible/roles/docker/handlers/main.yml

```diff
@@ -1,3 +1,39 @@
 ---
 - name: restart docker
-  service: name=docker state=restarted
+  command: /bin/true
+  notify:
+    - stop docker
+    - delete docker0
+    - start docker
+
+- name: start docker
+  command: /bin/true
+  notify:
+    - docker graph driver
+    - docker fix start on btrfs
+    - docker start service
+
+- name: stop docker
+  service: name=docker state=stopped
+  when: '"docker0" in ansible_interfaces'
+
+- name: delete docker0
+  command: ip link delete docker0
+  when: '"docker0" in ansible_interfaces'
+
+  # TODO: work only on default installation...
+- name: docker graph driver 
+  command: findmnt -no FSTYPE -T /var/lib/docker
+  register: docker_driver
+  changed_when: false
+
+  # TODO: fail if no --selinux-enabled in 'OPTIONS'
+- name: docker fix start on btrfs
+  replace:
+    dest="{{ docker_config_dir }}/docker"
+    regexp="^(OPTIONS=.*)(--selinux-enabled(?!=false)([^\s\"\']*))"
+    replace="\1--selinux-enabled=false"
```

Comment:
do you need a space before the --selinux in case there was no trailing space previously?